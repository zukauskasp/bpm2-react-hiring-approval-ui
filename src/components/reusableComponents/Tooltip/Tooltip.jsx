import React from "react";
import './Tooltip.scss';
import PropTypes from 'prop-types';

const Message = ({ messageType, text, position, ...otherPorps }) => {
    return <div className={`tooltip tooltip-${messageType} tooltip-${position}`} {...otherPorps}>{text} </div>
}

Message.propTypes = {
    messageType: PropTypes.oneOf(["error", "success", "default"]),
    position: PropTypes.oneOf(["bottom-left", "bottom", "bottom-right", "left", "top-left", "top", "top-right", "right"]),
    text: PropTypes.string,
}

Message.defaultProps = {
    messageType: "default",
    text: "",
    position: "top-left"
}

export default Message;