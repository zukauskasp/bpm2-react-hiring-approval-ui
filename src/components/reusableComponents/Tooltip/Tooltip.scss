.tooltip {
    position: relative;
    left: 0;
    top: 110%;
    color: white;
    min-height: 48px;
    font-size: 14px;
    width: 280px;
    padding: 0 .8rem;
    margin-top: 10px;
    padding: 15px;

    &:before {
        content: "\A";
        position: absolute;
        width: 0;
        height: 0;
    }
}

.tooltip-default {
    background: #003755;

    &:before {
        border: 10px solid #003755;
    }
}

.tooltip-error {
    background: #FBB273;

    &:before {
        border: 10px solid #FBB273;
    }
}

.tooltip-success {
    background: #09B89D;

    &:before {
        border: 10px solid #09B89D;
    }

}

.tooltip-top-left:before {
    bottom: 100%;
    border-top: none;
    border-left: 10px solid transparent;
    border-right: 10px solid transparent;

}

.tooltip-top:before {
    bottom: 100%;
    left: 45%;
    border-left: 10px solid transparent;
    border-right: 10px solid transparent;
    border-top: none;
}

.tooltip-top-right:before {
    bottom: 100%;
    left: 90%;
    border-top: none;
    border-left: 10px solid transparent;
    border-right: 10px solid transparent;
}


.tooltip-bottom-left:before {
    bottom: -20%;
    border-bottom: none;
    border-left: 10px solid transparent;
    border-right: 10px solid transparent;
}

.tooltip-bottom:before {
    bottom: -20%;
    left: 45%;
    border-bottom: none;
    border-left: 10px solid transparent;
    border-right: 10px solid transparent;
}

.tooltip-bottom-right:before {
    bottom: -20%;
    left: 90%;
    border-bottom: none;
    border-left: 10px solid transparent;
    border-right: 10px solid transparent;
}

.tooltip-right:before {
    bottom: 26%;
    left: 100%;
    border-right: none;
    border-top: 10px solid transparent;
    border-bottom: 10px solid transparent;
}

.tooltip-left:before {
    bottom: 26%;
    right: 100%;
    border-left: none;
    border-top: 10px solid transparent;
    border-bottom: 10px solid transparent;
}