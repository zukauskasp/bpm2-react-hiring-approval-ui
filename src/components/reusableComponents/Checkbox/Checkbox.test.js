import React from 'react';
import {shallow,mount} from 'enzyme';

import CustomCheckbox from './Checkbox';
import Tooltip from "../Tooltip/Tooltip";
import { FormControlLabel } from '@material-ui/core';

describe('Checkbox', ()=> {

    afterAll(() => {
        mount.cleanUp();
      });
      
    it('expect to match latest snapshot', () => {
        const wrapper =  mount(<CustomCheckbox errorMessage="Error message" label="label" onChange={()=>{}}/>);
        expect(wrapper.debug()).toMatchSnapshot();
    });

    it('expect to render checkbox component', () => {
        expect(shallow(<CustomCheckbox onChange={()=>{}}/>).length).toEqual(1);
    }); 

    it('expect to display error message', () => {
        const wrapper = shallow(<CustomCheckbox errorMessage="Error message" onChange={()=>{}}/>);
        expect(wrapper.find(Tooltip).length).toEqual(1);
    });

    it('expect to not display error message', () => {
        const wrapper = shallow(<CustomCheckbox onChange={()=>{}}/>);
        expect(wrapper.find(Tooltip).length).toEqual(0);
    });

    it('expect to display label', () => {
        const wrapper = shallow(<CustomCheckbox label="custom label" onChange={()=>{}}/>)
        expect(wrapper.find(FormControlLabel).props().label.props.children).toEqual('custom label');
    });

    it('expect to not display label', () => {
        const wrapper = shallow(<CustomCheckbox onChange={()=>{}}/>)
        expect(wrapper.find(FormControlLabel).props().label.props.children.length).toEqual(0);
    });
});