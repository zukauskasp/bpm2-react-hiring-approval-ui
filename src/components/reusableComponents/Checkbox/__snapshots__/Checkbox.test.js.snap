// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox expect to match latest snapshot 1`] = `
"<CustomCheckbox errorMessage=\\"Error message\\" label=\\"label\\" onChange={[Function: onChange]} readonly={false} value={false}>
  <div>
    <WithStyles(ForwardRef(FormGroup)) row={true}>
      <ForwardRef(FormGroup) classes={{...}} row={true}>
        <div className=\\"MuiFormGroup-root MuiFormGroup-row\\">
          <WithStyles(ForwardRef(FormControlLabel)) control={{...}} label={{...}} className=\\"makeStyles-margin-4\\">
            <ForwardRef(FormControlLabel) classes={{...}} control={{...}} label={{...}} className=\\"makeStyles-margin-4\\">
              <label className=\\"MuiFormControlLabel-root makeStyles-margin-4\\">
                <WithStyles(ForwardRef(Checkbox)) className=\\"makeStyles-customPrimaryCheckbox-2\\" checked={false} onChange={[Function: onChange]} value={false} disabled={false}>
                  <ForwardRef(Checkbox) classes={{...}} className=\\"makeStyles-customPrimaryCheckbox-2\\" checked={false} onChange={[Function: onChange]} value={false} disabled={false}>
                    <WithStyles(ForwardRef(SwitchBase)) type=\\"checkbox\\" classes={{...}} color=\\"secondary\\" inputProps={{...}} icon={{...}} checkedIcon={{...}} className=\\"makeStyles-customPrimaryCheckbox-2\\" checked={false} onChange={[Function: onChange]} value={false} disabled={false}>
                      <ForwardRef(SwitchBase) classes={{...}} type=\\"checkbox\\" color=\\"secondary\\" inputProps={{...}} icon={{...}} checkedIcon={{...}} className=\\"makeStyles-customPrimaryCheckbox-2\\" checked={false} onChange={[Function: onChange]} value={false} disabled={false}>
                        <WithStyles(ForwardRef(IconButton)) component=\\"span\\" className=\\"PrivateSwitchBase-root-19 MuiCheckbox-root MuiCheckbox-colorSecondary makeStyles-customPrimaryCheckbox-2\\" disabled={false} tabIndex={{...}} role={[undefined]} onFocus={[Function: handleFocus]} onBlur={[Function: handleBlur]} color=\\"secondary\\">
                          <ForwardRef(IconButton) classes={{...}} component=\\"span\\" className=\\"PrivateSwitchBase-root-19 MuiCheckbox-root MuiCheckbox-colorSecondary makeStyles-customPrimaryCheckbox-2\\" disabled={false} tabIndex={{...}} role={[undefined]} onFocus={[Function: handleFocus]} onBlur={[Function: handleBlur]} color=\\"secondary\\">
                            <WithStyles(ForwardRef(ButtonBase)) className=\\"MuiIconButton-root PrivateSwitchBase-root-19 MuiCheckbox-root MuiCheckbox-colorSecondary makeStyles-customPrimaryCheckbox-2 MuiIconButton-colorSecondary\\" centerRipple={true} focusRipple={true} disabled={false} component=\\"span\\" tabIndex={{...}} role={[undefined]} onFocus={[Function: handleFocus]} onBlur={[Function: handleBlur]}>
                              <ForwardRef(ButtonBase) classes={{...}} className=\\"MuiIconButton-root PrivateSwitchBase-root-19 MuiCheckbox-root MuiCheckbox-colorSecondary makeStyles-customPrimaryCheckbox-2 MuiIconButton-colorSecondary\\" centerRipple={true} focusRipple={true} disabled={false} component=\\"span\\" tabIndex={{...}} role={[undefined]} onFocus={[Function: handleFocus]} onBlur={[Function: handleBlur]}>
                                <span className=\\"MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-19 MuiCheckbox-root MuiCheckbox-colorSecondary makeStyles-customPrimaryCheckbox-2 MuiIconButton-colorSecondary\\" onBlur={[Function]} onClick={[undefined]} onFocus={[Function]} onKeyDown={[Function]} onKeyUp={[Function]} onMouseDown={[Function]} onMouseLeave={[Function]} onMouseUp={[Function]} onDragLeave={[Function]} onTouchEnd={[Function]} onTouchMove={[Function]} onTouchStart={[Function]} tabIndex={{...}} role={[undefined]} aria-disabled={false}>
                                  <span className=\\"MuiIconButton-label\\">
                                    <input autoFocus={[undefined]} checked={false} defaultChecked={[undefined]} className=\\"PrivateSwitchBase-input-22\\" disabled={false} id={[undefined]} name={[undefined]} onChange={[Function: handleInputChange]} readOnly={[undefined]} required={[undefined]} tabIndex={[undefined]} type=\\"checkbox\\" value={false} data-indeterminate={false} />
                                    <ForwardRef fontSize=\\"default\\">
                                      <WithStyles(ForwardRef(SvgIcon)) fontSize=\\"default\\">
                                        <ForwardRef(SvgIcon) classes={{...}} fontSize=\\"default\\">
                                          <svg className=\\"MuiSvgIcon-root\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" color={[undefined]} aria-hidden=\\"true\\" role=\\"presentation\\">
                                            <path d=\\"M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z\\" />
                                          </svg>
                                        </ForwardRef(SvgIcon)>
                                      </WithStyles(ForwardRef(SvgIcon))>
                                    </ForwardRef>
                                  </span>
                                  <NoSsr>
                                    <WithStyles(memo) center={true}>
                                      <ForwardRef(TouchRipple) classes={{...}} center={true}>
                                        <span className=\\"MuiTouchRipple-root\\">
                                          <TransitionGroup component={{...}} exit={true} childFactory={[Function: childFactory]} />
                                        </span>
                                      </ForwardRef(TouchRipple)>
                                    </WithStyles(memo)>
                                  </NoSsr>
                                </span>
                              </ForwardRef(ButtonBase)>
                            </WithStyles(ForwardRef(ButtonBase))>
                          </ForwardRef(IconButton)>
                        </WithStyles(ForwardRef(IconButton))>
                      </ForwardRef(SwitchBase)>
                    </WithStyles(ForwardRef(SwitchBase))>
                  </ForwardRef(Checkbox)>
                </WithStyles(ForwardRef(Checkbox))>
                <WithStyles(ForwardRef(Typography)) component=\\"span\\" className=\\"MuiFormControlLabel-label\\">
                  <ForwardRef(Typography) classes={{...}} component=\\"span\\" className=\\"MuiFormControlLabel-label\\">
                    <span className=\\"MuiTypography-root MuiFormControlLabel-label MuiTypography-body1\\">
                      <WithStyles(ForwardRef(Typography)) component=\\"p\\" className=\\"makeStyles-customLabel-1\\">
                        <ForwardRef(Typography) classes={{...}} component=\\"p\\" className=\\"makeStyles-customLabel-1\\">
                          <p className=\\"MuiTypography-root makeStyles-customLabel-1 MuiTypography-body1\\">
                            label
                          </p>
                        </ForwardRef(Typography)>
                      </WithStyles(ForwardRef(Typography))>
                    </span>
                  </ForwardRef(Typography)>
                </WithStyles(ForwardRef(Typography))>
              </label>
            </ForwardRef(FormControlLabel)>
          </WithStyles(ForwardRef(FormControlLabel))>
        </div>
      </ForwardRef(FormGroup)>
    </WithStyles(ForwardRef(FormGroup))>
    <Message text=\\"Error message\\" messageType=\\"error\\" position=\\"top-left\\">
      <div className=\\"tooltip tooltip-error tooltip-top-left\\">
        Error message
         
      </div>
    </Message>
  </div>
</CustomCheckbox>"
`;
